faker/providers/file/__init__.py

```diff
@@ -269,12 +270,28 @@ def file_path(
         If ``absolute`` is ``True`` (default), the generated path starts with
         ``/`` and is absolute. Otherwise, the generated path is relative.
 
+        If used, ``extension`` can be either a string, forcing that extension, a
+        list of strings (one will be picked at random), or an empty list (the
+        path will have no extension). Default behaviour is the same as |file_name|
+
         :sample: size=10
         :sample: depth=3
         :sample: depth=5, category='video'
         :sample: depth=5, category='video', extension='abcdef'
+        :sample: extension=[]
+        :sample: extension=["a", "bc", "def"]
         """
+        no_extension = False
+        if isinstance(extension, list):
```

Comment:
I'd like for `extension` to also possibly a tuple.
We could type the argument with `extension: Optional[Sequence[str]]`, and then check for `not isinstance(extension, str)` instead:

```suggestion
        if not isinstance(extension, str):
```

I would also change the behavior of `file_name`, so that if the empty string is passed, no extension and no dot are added. This would allow us to pass `""` to `file_name` and avoid stripping it out later.

Reply:
Done - also checking for None